driver/src/driver/deposit_driver.rs

```diff
@@ -296,45 +284,42 @@ mod tests {
             NUM_TOKENS,
         );
 
-        let contract = SnappContractMock::default();
+        let mut contract = MockSnappContract::default();
         contract
-            .get_current_deposit_slot
-            .given(())
-            .will_return(Ok(slot));
+            .expect_get_current_deposit_slot()
+            .return_const(Ok(slot));
         contract
-            .has_deposit_slot_been_applied
-            .given(slot)
-            .will_return(Ok(false));
+            .expect_has_deposit_slot_been_applied()
+            .with(eq(slot))
+            .return_const(Ok(false));
         contract
-            .has_deposit_slot_been_applied
-            .given(slot - 1)
-            .will_return(Ok(true));
+            .expect_has_deposit_slot_been_applied()
+            .with(eq(slot - 1))
+            .return_const(Ok(true));
 
         contract
-            .creation_timestamp_for_deposit_slot
-            .given(slot)
-            .will_return(Ok(U256::from(10)));
+            .expect_creation_timestamp_for_deposit_slot()
+            .with(eq(slot))
+            .return_const(Ok(U256::from(10)));
         contract
-            .get_current_block_timestamp
-            .given(())
-            .will_return(Ok(U256::from(200)));
+            .expect_get_current_block_timestamp()
+            .return_const(Ok(U256::from(200)));
 
         contract
-            .deposit_hash_for_slot
-            .given(slot)
-            .will_return(Ok(H256::zero()));
+            .expect_deposit_hash_for_slot()
+            .with(eq(slot))
+            .return_const(Ok(H256::zero()));
         contract
-            .get_current_state_root
-            .given(())
-            .will_return(Ok(state_hash));
+            .expect_get_current_state_root()
+            .return_const(Ok(state_hash));
 
-        let db = DbInterfaceMock::new();
-        db.get_deposits_of_slot
-            .given(U256::one())
-            .will_return(Ok(deposits));
-        db.get_balances_for_state_root
-            .given(state_hash)
-            .will_return(Ok(state));
+        let mut db = MockDbInterface::new();
+        db.expect_get_deposits_of_slot()
+            .with(eq(U256::one()))
+            .return_const(Ok(deposits));
+        db.expect_get_balances_for_state_root()
+            .with(eq(state_hash))
+            .return_const(Ok(state));
```

Comment:
what is the `eq` for?

Reply:
It's a matcher that will consider this method as being called when the argument is equal to state_hash

Reply:
Interesting. So then if we call this function with any other state_root, the mock would treat that as some kind of Error. Makes sense thanks.

Reply:
Yes, there are other matchers such as always (they match anything) and you can even provide a matching function that can inspect the argument and return a boolean if it should match or not.